doc/system-administration.md

```diff
@@ -179,10 +179,91 @@ There should be an entry for `certbot renew`. That command may be run manually a
+```
+docker: Error response from daemon: failed to create endpoint driver-app on network bridge: iptables failed: iptables --wait -t filter -A DOCKER ! -i docker0 -o docker0 -p tcp -d 172.17.0.2 --dport 4000 -j ACCEPT: iptables: No chain/target/match by that name.
+```
+Restarting Docker can be performed by running:
+```
+service docker restart
+```
+After Docker is running, `driver-app` will then need to be restarted again.
+
+
+### 3. Ensure Redis is running
+
+Redis needs to have open permissions due to its usage via Windshaft. It is therefore paramount that network traffic is locked down via `ufw` or Security Groups. If the machine is running without a firewall in place for an extended period of time, there is a chance an automated system will be able to write data to it and cause a corrupt state. Once it's verified that `ufw` is running, check for any peculiar logs in `/var/log/redis/redis-server.log`, and check if Redis is running with the command:
+```
+sudo service redis-server status
+```
+If Redis is down, or the logs are showing any kind of disk persistence errors, it may be restarted with the command:
+```
+sudo service redis-server restart
+```
+There may also be problems with restarting `driver-app` in the above step if Redis is not behaving properly. If the App machine is showing Redis-related errors in `/var/log/upstart/driver-app.log`, then be sure to restart Redis before attempting to restart `driver-app`.
+
+
+### 4. Ensure Celery is running
+
+Celery is responsible for running all asynchronous tasks within the application. This includes CSV exports, as well as running black spot calculations. To check if Celery is running, log into the Celery machine, and run the following command:
+```
+service driver-celery status
+```
+Also check the log at `/var/log/upstart/driver-celery.log`. If there are any errors related to Docker, perform the same instructions related to restarting Docker from the `driver-app` step above, and then restart Celery with:
+
+```
+service driver-celery restart
+```
+
+
+### 5. Verify the application is functioning
+
+All services should now be running, reporting valid status, and showing good log messages: ufw and Docker are running on each machine, `driver-app` is running on the App machine, `driver-celery` is running on the Celery machine, and Redis is running on the Database machine.
+
+Log into the web application, view the dashboard, the record list page, and the map page to ensure records are displaying properly in each of the pages. Then attempt to perform a CSV export on the map page. If it completes successfully, then the app is back in a functional state. If it times out, log into the Celery machine to see if jobs are processing. View tail the log file at: `/var/log/upstart/driver-celery.log`. Hopefully there are some jobs processing, demonstrated by recent log messages such as:
```

Comment:
"View tail the log file" reads a little awkwardly to me, how about 
> Follow the log file with: `tail -f /var/log/upstart/driver-celery.log`
```suggestion
Log into the web application, view the Dashboard, the Record List page, and the Map page to ensure records are displaying properly in each of the pages. Then attempt to perform a CSV export on the Map page. If it completes successfully, then the app is back in a functional state. If it times out, log into the Celery machine to see if jobs are processing. Follow the log file with:
\```
tail -f /var/log/upstart/driver-celery.log
\```
Hopefully there are some jobs processing, demonstrated by recent log messages such as:
```
(Not sure how to embed triple-backticks inside a triple-backtick...)